### PR TITLE
[TM] Add spec for DevSettings

### PR DESCRIPTION
**Pull request by jarvisluong**
## Summary

Part of #24875, adds a spec for DevSettings.

## Changelog

[General] [Added] - TM spec for DevSettings

## Test Plan

Flow passes

### CODE REVIEW

File: Libraries/NativeModules/specs/NativeDevSettings.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  +reload: () => void;
+  +setHotLoadingEnabled: (isHotLoadingEnabled: boolean) => void;
+  +setIsDebuggingRemotely: (isDebugginRemotelyEnabled: boolean) => void;
+  +setLiveReloadEnabled: (isLiveReloadEnabled: boolean) => void;
+  +setProfilingEnabled: (isProfilingEnabled: boolean) => void;
+  +toggleElementInspector: () => void;
+  +setIsShakeToShowDevMenuEnabled?: (enabled: boolean) => void;
+}
+
+export default TurboModuleRegistry.get<Spec>('DevSettings');
```
Comment (fkgozali) [inline Libraries/NativeModules/specs/NativeDevSettings.js]: could you use `getEnforcing` instead of `get`?

Comment (jarvisluong) [reply Libraries/NativeModules/specs/NativeDevSettings.js]: yes, my bad that I forgot this.

File: Libraries/NativeModules/specs/NativeDevSettings.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import * as TurboModuleRegistry from '../../TurboModule/TurboModuleRegistry';
+
+export interface Spec extends TurboModule {
+  +reload: () => void;
+  +setHotLoadingEnabled: (isHotLoadingEnabled: boolean) => void;
+  +setIsDebuggingRemotely: (isDebuggingRemotelyEnabled: boolean) => void;
+  +setLiveReloadEnabled: (isLiveReloadEnabled: boolean) => void;
+  +setProfilingEnabled: (isProfilingEnabled: boolean) => void;
+  +toggleElementInspector: () => void;
+  +setIsShakeToShowDevMenuEnabled?: (enabled: boolean) => void;
```
Comment (thymikee) [inline Libraries/NativeModules/specs/NativeDevSettings.js]: Should we mark this as iOS only with a comment `// iOS only` like we do for Android (`// Android only`)?

Comment (fkgozali) [reply Libraries/NativeModules/specs/NativeDevSettings.js]: Yeah, and this should just be required. I'll adjust it when I import this.

**Pull request closed without merging**